src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -1322,6 +1324,44 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    void TerminalWindow::_WindowSizeChanged(const IInspectable&, winrt::Microsoft::Terminal::Control::WindowSizeChangedEventArgs args)
+    {
+        til::size cellCount{ args.Width(), args.Height() };
+        const auto dpi = GetDpiForWindow(_hwnd);
+        const auto settings{ TerminalSettings::CreateWithNewTerminalArgs(_settings, nullptr, nullptr) };
+        auto pixelSize = TermControl::GetProposedDimensions(settings.DefaultSettings(), dpi, cellCount.width, cellCount.height);

```

Comment:
This might be problematic. The default settings for the default profile might not be what's currently active in that terminal window. Consider the scenario where you ctrl+mouse to increase the font size. The settings object doesn't actually know about that change, only the `TermControl` raising the event does.

I bet this could be solved by having the TermControl itself handle the `WindowSizeChanged` from the `ControlCore` (instead of straight bubbling it). The TermControl would get the size in chars, use that to figure out how big it wants to be in DIPs, then raise the size in DIPs up to the `TerminalPage`